internal/reader/onchain_prices_reader.go

```diff
@@ -84,5 +94,67 @@ func (pr *OnchainTokenPricesReader) GetTokenPricesUSD(
 	return prices, nil
 }
 
+func (pr *OnchainTokenPricesReader) getFeedDecimals(ctx context.Context, token types.Account) (*uint8, error) {
+	var decimals *uint8
+	if err :=
+		pr.ContractReader.GetLatestValue(
+			ctx,
+			consts.ContractNamePriceAggregator,
+			consts.MethodNameGetDecimals,
+			primitives.Unconfirmed,
+			nil,
+			&decimals,
```

Comment:
Ditto here

Reply:
It was passing integration tests without issues. I fixed it to be a non pointer anyways.